### PR TITLE
Added CRUD functionalities for Product Categories

### PR DESCRIPTION
**Pull request by Foca22**
Please review the CRUD functionalities for Product Categories

### CODE REVIEW

File: src/main/java/com/project/store/dto/category/ProductCategoryDto.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.project.store.dto.category;
+
+import javax.validation.constraints.NotEmpty;
+
+public class ProductCategoryDto {
+
+    private Integer id;
+
+    @NotEmpty(message = "product.category.name.cannot.be.empty")
+    String name;
```
Comment (diana-stoica-ub) [inline src/main/java/com/project/store/dto/category/ProductCategoryDto.java]: private

File: src/main/java/com/project/store/dto/category/ProductCategoryDto.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @NotEmpty(message = "product.category.type.cannot.be.empty")
+    String type;
```
Comment (diana-stoica-ub) [inline src/main/java/com/project/store/dto/category/ProductCategoryDto.java]: private

File: src/main/java/com/project/store/exceptions/category/ProductCategoryExceptionMessages.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.project.store.exceptions.category;
+
+import org.springframework.http.HttpStatus;
+
+public enum ProductCategoryExceptionMessages {
+
+    PRODUCT_CATEGORY_NOT_FOUND("Product Category not found!", HttpStatus.NOT_FOUND.toString());
```
Comment (diana-stoica-ub) [inline src/main/java/com/project/store/exceptions/category/ProductCategoryExceptionMessages.java]: I would use only one enum for all the errors, so I would merge this one with the one created for Customer

File: src/main/java/com/project/store/model/category/ProductCategory.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import javax.persistence.*;
+
+@Entity
+@Table(name = "product_categories")
+public class ProductCategory {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "product_category_id", updatable = false, nullable = false)
+    Integer id;
```
Comment (diana-stoica-ub) [inline src/main/java/com/project/store/model/category/ProductCategory.java]: private fields

File: src/main/java/com/project/store/dto/category/ProductCategoryDto.java (hunk 2 above)
Comment (Foca22) [reply src/main/java/com/project/store/dto/category/ProductCategoryDto.java]: Name and Type fields were made private

File: src/main/java/com/project/store/exceptions/category/ProductCategoryExceptionMessages.java (hunk 3 above)
Comment (Foca22) [reply src/main/java/com/project/store/exceptions/category/ProductCategoryExceptionMessages.java]: Exception messages from Customer and ProductCategories were merged into a single enum called Exception messages

File: src/main/java/com/project/store/model/category/ProductCategory.java (hunk 4 above)
Comment (Foca22) [reply src/main/java/com/project/store/model/category/ProductCategory.java]: Class fields for ProductCategories were made private

**Review by diana-stoica-ub — APPROVED** [review]

**Pull request merged**